Error calling GetJMSCorrelationID when empty

When receive a message with empty correlationId the mqmd.CorrelId property value is "000000000000000000000000000000000000000000000000" wich throws this error :

```
panic: runtime error: index out of range [-1] [recovered]
        panic: runtime error: index out of range [-1]

at github.com/ibm-messaging/mq-golang-jms20/mqjms/MessageImpl.go:179 
```

The current code throws error because all bytes are zero and tries to access to correlIDBytes[-1]
```golang
if realLength > 0 {
	for correlIDBytes[realLength-1] == 0 { // error line
		realLength--
	}
}
```

--- a/getbycorrelid_test.go
+++ b/getbycorrelid_test.go
@@ -202,4 +202,9 @@ func TestCorrelIDParsing(t *testing.T) {
 	msg.SetJMSCorrelationID(testCorrel)
 	assert.Equal(t, testCorrel, msg.GetJMSCorrelationID())
 
+	// Empty correlationID
+	testCorrel = "000000000000000000000000000000000000000000000000"
+	msg.SetJMSCorrelationID(testCorrel)
+	assert.Equal(t, "", msg.GetJMSCorrelationID())
+
 }

--- a/mqjms/MessageImpl.go
+++ b/mqjms/MessageImpl.go
@@ -175,10 +175,8 @@ func (msg *MessageImpl) GetJMSCorrelationID() string {
 		// Here we identify any padding zero bytes to trim off so that we can try
 		// to turn it back into a string.
 		realLength := len(correlIDBytes)
-		if realLength > 0 {
-			for correlIDBytes[realLength-1] == 0 {
-				realLength--
-			}
+		for realLength > 0 && correlIDBytes[realLength-1] == 0 {
+			realLength--
 		}
 
 		// Attempt to decode the content back into a string.

